src/bundle.ts

```diff
@@ -82,11 +90,23 @@ export class BuildBundler extends Transform {
     const manifest = await this._bundler.generateManifest(
         bundleEntrypoints.map(f => urlFromPath(this.config.root, f)), strategy);
     const docCollection = await this._bundler.bundle(manifest);
+
+    // Remove the bundled files from the file map so they are not emitted later.
+    this._unmapBundledFiles(manifest);
+
     const contentsMap = new Map();
     for (const bundleName of docCollection.keys()) {
       contentsMap.set(
           bundleName, parse5.serialize(docCollection.get(bundleName).ast));
     }
     return contentsMap;
   }
+
+  _unmapBundledFiles(manifest: BundleManifest) {
+    for (const bundle of manifest.bundles.values()) {
+      for (const filename of bundle.files) {
+        this.files.delete(filename);
```

Comment:
I can't comment on the `this.files` definition above, but please document what `this.files` state represents.  In all other streams it represents a cache of all files, but now it's contents actually dictate the stream output. This would be useful to explain somewhere for later readers.